### PR TITLE
Fixed PR changes, modified coingecko route, implemented array destruc…

### PR DESCRIPTION
**Pull request by arbreton**
…turing

### CODE REVIEW

**Comment by aminlatifi** [comment]
@arbreton You have forgotten to merge latest version of develop branch to your f_coinGeckoAPI branch

File: src/services/conversionRates/getConversionRatesService.js [hunk 1]
```diff
@@ -50,45 +50,18 @@ const _getRatesCoinGecko = async (
   const promises = ratesToGet.map(async r => {
     const rateSymbolInner = stableCoins.includes(r) ? 'USD' : r;
     if (rateSymbolInner !== rateSymbol) {
-      const today = new Date(new Date());
-      const todayUTC = today.setUTCHours(0, 0, 0, 0);
-      const todayRounded = Math.round(todayUTC / 1000);
-      const timestampRounded = Math.round(timestampMS / 1000);
-      const differenceMillis = todayRounded - timestampRounded;
-      let differenceDays = 0;
+      const timestampTo = Math.round(timestampMS / 1000);
```
Comment (aminlatifi) [inline src/services/conversionRates/getConversionRatesService.js:53]: set timestampTo later than timestampMS as timestampFrom is sooner than it.

File: src/services/conversionRates/getConversionRatesService.js [hunk 2]
```diff
@@ -50,45 +50,18 @@ const _getRatesCoinGecko = async (
...
-          cur = prices[i];
-          priceTime = Math.round(cur[0] / 1000);
-          difference = Math.abs(timestampRounded - priceTime);
-          if (difference < bestDifference) {
-            bestDifference = difference;
-            bestIndex = i;
-          }
-        }
-        const bestPrices = prices[bestIndex];
+        const bestPrices = prices[0];
```
Comment (aminlatifi) [inline src/services/conversionRates/getConversionRatesService.js:64]: The first price returned is not the best (closest one). Iterate over them to find the best one.

File: src/services/conversionRates/getConversionRatesService.js [hunk 3]
```diff
@@ -50,45 +50,18 @@ const _getRatesCoinGecko = async (
...
         const someIndex = 1;
         bestPrice = bestPrices[someIndex];
```
Comment (aminlatifi) [inline src/services/conversionRates/getConversionRatesService.js:66]: I suggest `const [timestamp, price] = bestPrices`. It's array destructing

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/services/conversionRates/getConversionRatesService.js (hunk 1 above)
Comment (arbreton) [reply src/services/conversionRates/getConversionRatesService.js:53]: Checked it, timestampTo is in the future from timestampFrom now

File: src/services/conversionRates/getConversionRatesService.js (hunk 2 above)
Comment (arbreton) [reply src/services/conversionRates/getConversionRatesService.js:64]: Re implemented timeframe refination

File: src/services/conversionRates/getConversionRatesService.js (hunk 3 above)
Comment (arbreton) [reply src/services/conversionRates/getConversionRatesService.js:66]: I see it's a great alternative to destructure the object, implemented it

**Comment by arbreton** [comment]
Updated PR with requested changes @aminlatifi 🙌

**Review by aminlatifi — APPROVED** [review]

**Comment by aminlatifi** [comment]
LGTM!

**Pull request merged**